src/Factory/MigrationFactory.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace BowlOfSoup\CouchbaseMigrationsBundle\Factory;
+
+use BowlOfSoup\CouchbaseMigrationsBundle\Migration\AbstractMigration;
+use ReflectionClass;
+use Symfony\Component\Finder\SplFileInfo;
+
+class MigrationFactory
+{
+    /** @var \BowlOfSoup\CouchbaseMigrationsBundle\Factory\ClusterFactory */
+    private $clusterFactory;
+
+    /**
+     * @param \BowlOfSoup\CouchbaseMigrationsBundle\Factory\ClusterFactory $clusterFactory
+     */
+    public function __construct(ClusterFactory $clusterFactory)
+    {
+        $this->clusterFactory = $clusterFactory;
+    }
+
+    /**
+     * @param \Symfony\Component\Finder\SplFileInfo $file
+     *
+     * @return \BowlOfSoup\CouchbaseMigrationsBundle\Migration\AbstractMigration
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function createByFile(SplFileInfo $file): AbstractMigration
+    {
+        if ($file->getExtension() !== 'php') {
```

Comment:
Instead of this check, only search for `*.php` files with the `Finder`.

```
$finder->files()->in($this->migrationsDirectory)->name('*.php');
```

Reply:
I added the .php extension to the find criteria in the MigrateCommand.
However, to make sure this unit will never fail if being used wrong in the future, i left the exception in place.